### PR TITLE
Implement last changes from #344

### PR DESCRIPTION
**Pull request by cgranade**
This PR completes work started with https://github.com/microsoft/QuantumLibraries/issues/322 and #212, and as reviewed at #344. After this PR, we should be good to merge the relevant feature branch to `main`.

### CODE REVIEW

File: Standard/src/Arithmetic/Asserts.qs [hunk 1]
```diff
@@ -32,14 +32,15 @@ namespace Microsoft.Quantum.Arithmetic {
...
+    /// # Example
     /// Suppose that the `qubits` register encodes a 3-qubit quantum state
     /// $\ket{\psi}=\sqrt{1/8}\ket{0}+\sqrt{7/8}\ket{6}$ in little-endian format.
     /// This means that the number states $\ket{0}\equiv\ket{0}\ket{0}\ket{0}$
     /// and $\ket{6}\equiv\ket{0}\ket{1}\ket{1}$. Then the following asserts succeed:
-    /// - `AssertProbInt(0,0.125,qubits,10e-10);`
-    /// - `AssertProbInt(6,0.875,qubits,10e-10);`
+    /// ```qsharp
+    /// AssertProbInt(0,0.125,qubits,10e-10);
+    /// AssertProbInt(6,0.875,qubits,10e-10);
```
Comment (msoeken) [inline Standard/src/Arithmetic/Asserts.qs:42]:
```suggestion
    /// AssertProbInt(0, 0.125, qubits, 10e-10);
    /// AssertProbInt(6, 0.875, qubits, 10e-10);
```

File: Standard/src/Preparation/Arbitrary.qs [hunk 2]
```diff
@@ -12,101 +12,110 @@ namespace Microsoft.Quantum.Preparation {
...
+    /// prepares an state on that register described by the given coefficients.
     ///
-    /// The returned operation $U$ prepares an arbitrary quantum
+    /// # Description
+    /// This operation prepares an arbitrary quantum
     /// state $\ket{\psi}$ with complex coefficients $r_j e^{i t_j}$ from
-    /// the $n$-qubit computational basis state $\ket{0...0}$.
+    /// the $n$-qubit computational basis state $\ket{0 \cdots 0}$.
+    /// In particular, the action of this operation can be simulated by the
+    /// a unitary transformation $U$ which acts on the all-zeros state as
```
Comment (msoeken) [inline Standard/src/Preparation/Arbitrary.qs:74]:
```suggestion
    /// a unitary transformation $U$ that acts on the all-zeros state as
```

File: Standard/src/Preparation/Deprecated.qs [hunk 3]
```diff
@@ -92,4 +92,236 @@ namespace Microsoft.Quantum.Preparation {
...
+    /// $H$ for `PauliX`, $HS$ for `PauliY`, $I$ for `PauliZ` and
+    /// <xref:microsoft.quantum.preparation.preparesinglequbitidentity> for `PauliI`.
+    ///
+    /// # Input
+    /// ## basis
+    /// A Pauli operator $P$.
+    /// ## qubit
+    /// A qubit to be prepared.
+    @Deprecated("Microsoft.Quantum.Preparation.PreparePauliEigenstate")
+    operation PrepareQubit (basis : Pauli, qubit : Qubit) : Unit {
```
Comment (msoeken) [inline Standard/src/Preparation/Deprecated.qs:115]:
```suggestion
    operation PrepareQubit(basis : Pauli, qubit : Qubit) : Unit {
```

File: Standard/src/Preparation/Deprecated.qs [hunk 4]
```diff
@@ -92,4 +92,236 @@ namespace Microsoft.Quantum.Preparation {
...
+    ///     set complexNumbers[idx] = ComplexPolar(amplitudes[idx], phases[idx]);
+    /// }
+    /// let op = StatePreparationComplexCoefficients(complexNumbers);
+    /// using (qubits = Qubit[2]) {
+    ///     let qubitsLE = LittleEndian(qubits);
+    ///     op(qubitsLE);
+    /// }
+    /// ```
+    @Deprecated("Microsoft.Quantum.Preparation.PrepareArbitraryStateCP")
+    function StatePreparationComplexCoefficients (coefficients : ComplexPolar[]) : (LittleEndian => Unit is Adj + Ctl) {
```
Comment (msoeken) [inline Standard/src/Preparation/Deprecated.qs:218]:
```suggestion
    function StatePreparationComplexCoefficients(coefficients : ComplexPolar[]) : (LittleEndian => Unit is Adj + Ctl) {
```

**Review by msoeken — APPROVED** [review]
Looks good to me @cgranade. Thanks a lot for all your work on this.

**Comment by cgranade** [comment]
Thanks for the review, @msoeken! I'll go on and merge this in now, then, and get the final PR to `main` set up so that we can do a final review to include this in the next release.

**Pull request merged**